runtime/browser/media/media_capture_devices_dispatcher.cc

```diff
@@ -127,7 +128,10 @@ void XWalkMediaCaptureDevicesDispatcher::RequestPermissionToUser(
     const content::MediaResponseCallback& callback) {
   DCHECK(content::BrowserThread::CurrentlyOn(content::BrowserThread::UI));
 
-  if (request.security_origin.SchemeIs(url::kHttpScheme))
+  // This function may be called for a media request coming from
+  // from WebRTC/mediaDevices. These requests can't be made from HTTP.
+  if (request.security_origin.SchemeIs(url::kHttpScheme) &&
+      !utils::IsLocalHost(request.security_origin))
```

Comment:
Can we use `net::IsLocalhost`? (https://github.com/crosswalk-project/chromium-crosswalk/blob/master/net/base/net_util.h#L102)